node/src/protocol/consensus.rs

```diff
@@ -186,15 +186,15 @@ impl ConsensusProtocol for StartedState {
                                 "starting key generation as a part of the participant set"
                             );
                             let participants = contract_state.participants;
-                            let protocol = cait_sith::keygen::<Secp256k1>(
+                            let protocol = KeygenProtocol::new(
                                 &participants.keys().cloned().collect::<Vec<_>>(),
                                 me,
                                 contract_state.threshold,
                             )?;
                             Ok(NodeState::Generating(GeneratingState {
                                 participants,
                                 threshold: contract_state.threshold,
-                                protocol: Arc::new(RwLock::new(protocol)),
+                                protocol,
```

Comment:
should this be protocol.protocol?

Reply:
nope, generating state should not have direct access to the inner protocol since it is managed by the `KeygenProtocol` itself. The `KeygenProtocol` type was changed from `Arc<RwLock<_>>` to be its own type so it might've looked like it was still of type `Arc<RwLock<_>>` here but it's not